src/test_storageiface.py

```diff
@@ -0,0 +1,139 @@
+'''
+test_storageiface.py
+
+Basic unit testing of the storage interfaces
+'''
+
+import unittest
+import logging
+import configparser
+
+class TestStorage(unittest.TestCase):
+  '''Simple tests for the storage layers of the WOPI server'''
+
+  def __init__(self, *args, **kwargs):
+    '''One-off initialization of the test environment: create mock logging and import the library'''
+    super(TestStorage, self).__init__(*args, **kwargs)
+    loghandler = logging.FileHandler('/tmp/wopiserver.log')
+    loghandler.setFormatter(logging.Formatter(fmt='%(asctime)s %(name)s[%(process)d] %(levelname)-8s %(message)s',
+                                              datefmt='%Y-%m-%dT%H:%M:%S'))
+    log = logging.getLogger('wopiserver.test')
+    log.addHandler(loghandler)
+    log.setLevel(logging.DEBUG)
+    # read the configuration
+    config = configparser.ConfigParser()
+    try:
+      with open('wopiserver-test.conf') as fdef:
+        config.read_file(fdef)
+      storagetype = config.get('general', 'storagetype')
+      self.userid = config.get(storagetype, 'userid')
+      self.endpoint = config.get(storagetype, 'endpoint')
+    except (KeyError, configparser.NoOptionError):
+      print("Missing option or missing configuration, check the wopiserver-test.conf file")
+      raise
+
+    # this is taken from wopiserver.py::storage_layer_import
+    if storagetype in ['local', 'xroot', 'cs3']:
+      storagetype += 'iface'
```

Comment:
Can't we just add the iface in the config?

Reply:
Well, that's a matter of taste. I actually prefer to have the config talk about storage types, not interfaces which are an implementation detail.

Reply:
Ok, fine with me!